### DISCUSSION TITLE
Rate-Limitting Kong cann't connect Redis TLS/SSL

### DISCUSSION THREAD

**Discussion by romvarac** [Kong/kong#7296] [Ideas and feature requests]
Hi Team,
Configuration rate-limitting kong with redis policy, I config host, port, password, but have a error
```
[kong] init.lua:152 failed to auth Redis: closed
```
I using redis DigitalOccean Database, just connect by TLS/SSL.
So, how I could connect Redis. Thanks

**Comment by dndx** [comment]
Hello,

I do not believe the rate limit plugin currently supports connecting to Redis via TLS. Partially because such support is only available in enterprise version of Redis.

**Comment by jSherz** [comment]
As of [version 6 of Redis](https://redis.io/topics/encryption), SSL will be supported natively. Additionally, AWS Elasticache supports it now and other providers (as OP uses) can wrap it with Stunnel or similar tools.

Resty Redis does support SSL now: https://github.com/openresty/lua-resty-redis#connect and so this could be added by passing through an option for it.

**Comment by gabeio** [comment]
my rate-limiting redis ssl patch was just merged today, I assume that means it will go out in the next release, just wanted to share an update 👍.
